### ISSUE TITLE
File format request: only write LF on all platforms

### ISSUE THREAD

**Issue by adam-coster** [YoYoGames/GameMaker-Bugs#4820]
### Is your feature request related to a problem?

Windows-style line endings (CRLF) are an unfortunate legacy behavior that isn't required for pretty much any modern software to properly interpret line endings in a text file. Historically, GameMaker has written its project files to use CRLF on Windows.

Recently, the new format seems to have switched to using LF instead of CRLF. This is a welcome change! But it's creating a temporary headache while we reconfigure our Git settings and the settings of other tools, and I want to make sure that this change is stable.

### Describe the solution you'd like

I *think* it's the case that the new `yy` and `yyp` file formats now use POSIX-style LF endings instead of CRLF (due to those characters being invisible, and modern tools often automatically converting, it's hard to tell for sure), but I want to make an official request that:

1. GameMaker tools (ProjectTool, the IDE, etc) only ever write POSIX-style LF endings
2. The GameMaker team commit to ensuring that's true now and stays true for future updates

**Comment by stuckie** [comment]
We haven't actually made any conscious decision to do this, and it's not something we can particularly rely on since, as you've already noticed, git itself can change line endings between the two depending upon it's settings.
Even differing editors have a tendency to disagree and we certainly have source files that can end up with a mix of the two.
While I'd like to say it'll stay stable as it is now from the output we are writing, we can't guarantee it, nor what other external tooling will do to it in transit.

**Comment by adam-coster** [comment]
In that case y'all may want to put together some guidelines in the docs for how teams can avoid line-ending-based conflicts. It'd be useful in any case, but especially if the line endings generated by GameMaker are subject to change.

We're trying to keep things under control via this line in our `.gitattributes` file:

`* text eol=lf`

Though since that applies to *all* files, it would probably be better for users to be more specific, e.g.

```
*.gml text eol=lf
*.yy text eol=lf
*.yyp text eol=lf
*.json text eol=lf
```

**Comment by stuckie** [comment]
That may go well with #2922 so that new projects are setup properly from the outset.

**Comment by stuckie** [comment]
I'm going to close this now as the above changes got merged in with #2922 which should stop git messing with line endings.
